editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -2260,12 +2260,14 @@ void SpatialEditorViewport::_notification(int p_what) {
 		surface->connect("mouse_exited", this, "_surface_mouse_exit");
 		surface->connect("focus_entered", this, "_surface_focus_enter");
 		surface->connect("focus_exited", this, "_surface_focus_exit");
-		view_menu->set_flat(false);
-		view_menu->add_style_override("normal", editor->get_gui_base()->get_stylebox("Information3dViewport", "EditorStyles"));
-		view_menu->add_style_override("hover", editor->get_gui_base()->get_stylebox("Information3dViewport", "EditorStyles"));
-		view_menu->add_style_override("pressed", editor->get_gui_base()->get_stylebox("Information3dViewport", "EditorStyles"));
-		view_menu->add_style_override("focus", editor->get_gui_base()->get_stylebox("Information3dViewport", "EditorStyles"));
-		view_menu->add_style_override("disabled", editor->get_gui_base()->get_stylebox("Information3dViewport", "EditorStyles"));
+
+		// Needed to force the MenuButton to have the normal style.
+		view_menu->add_style_override("normal", editor->get_gui_base()->get_stylebox("normal", "Button"));
+		view_menu->add_style_override("hover", editor->get_gui_base()->get_stylebox("hover", "Button"));
+		view_menu->add_style_override("pressed", editor->get_gui_base()->get_stylebox("pressed", "Button"));
+		view_menu->add_style_override("focus", editor->get_gui_base()->get_stylebox("focus", "Button"));
+		view_menu->add_style_override("disabled", editor->get_gui_base()->get_stylebox("disabled", "Button"));
+
```

Comment:
the setup part where you add the style overrides is only executed, when the node enters the tree (on editor startup) changing the theme does not update button style...
Add the style override to the NOTIFICATION_THEME_CHANGED signal and it gets updated.